ably-ios/ARTRealtime.m

```diff
@@ -758,7 +768,11 @@ - (void)realtimeTransportClosed:(id<ARTRealtimeTransport>)transport {
 }
 
 - (void)realtimeTransportDisconnected:(id<ARTRealtimeTransport>)transport {
-    [self transition:ARTRealtimeDisconnected];
+    if (self.connection.state == ARTRealtimeClosing) {
+        [self transition:ARTRealtimeClosed];
```

Comment:
It needs to wait for the CLOSED action to transition to the Closed state. [RTN12a](http://docs.ably.io/client-lib-development-guide/features/#RTN12a)